core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1080,6 +1122,25 @@ class DAGScheduler(
     }
   }
 
+  // Visible for testing.
+  private[spark] def getTaskInputSizesFromShuffledRDD(tasks: Seq[Task[_]]): Map[Task[_], Long] = {
+    val taskInputSizeFromShuffledRDD = HashMap[Task[_], Long]()
+    tasks.foreach {
+      case task =>
+        val size =
+          parentSplitsInShuffledRDD(task.stageId, task.partitionId).map {
+            case parentSplits =>
+              parentSplits.map {
+                case (shuffleId, splits) =>
+                  splits.map(mapOutputTracker.getMapSizesByExecutorId(shuffleId, _)
+                    .flatMap(_._2.map(_._2)).sum).sum
```

Comment:
I think you can use `mapOutputTracker.getStatistics` here.

It also occurs to me that in general this could use the *total* input size for the task, but I guess spark isn't looking at that in general yet (though it probably could, from hadoop's `InputSplit.getLength()`).  Just something to keep in mind.